src/rebar_string.erl

```diff
@@ -36,6 +37,10 @@ join(Strings, Separator) -> string:join(Strings, Separator).
 split(Str, SearchPattern) when is_list(Str) -> string:split(Str, SearchPattern);
 split(Str, SearchPattern) when is_binary(Str) -> binary:split(Str, SearchPattern).
 lexemes(Str, SepList) -> string:tokens(Str, SepList).
+trim(Str) when is_list(Str) ->
+    re:replace(Str, "\\s+", "", [global, {return, list}]);
+trim(Str) when is_binary(Str) ->
+    list_to_binary(trim(binary_to_list(Str))).
```

Comment:
This is not unicode-safe; you want to use `unicode:characters_to_binary` and `unicode:characters_to_list` here to be equivalent to what `string` does, and possibly pass the `unicode` argument to the `re` module. However that won't work well with non-unicode strings in the first place.

Reply:
It is only for pre-20 and version numbers.

Reply:
I guess if it ends up being used for more general cases we'll find it sooner or later. It's not like the unicode stuff would have worked otherwise for string functions (although it did for `re`)

Reply:
And we are just a major version or 2 away from being able to remove it entirely.